workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/EntityGameObjectLinker.cs

```diff
@@ -21,9 +21,28 @@ public EntityGameObjectLinker(World world, WorkerSystem worker)
             entityManager = world.GetExistingManager<EntityManager>();
         }
 
-        public void LinkGameObjectToEntity(GameObject gameObject, Entity entity, EntityId spatialEntityId,
-            ViewCommandBuffer viewCommandBuffer)
+        public void LinkGameObjectToEntity(GameObject gameObject, Entity entity, ViewCommandBuffer viewCommandBuffer)
         {
+            bool hasSpatialEntityId = entityManager.HasComponent<SpatialEntityId>(entity);
+            bool isWorkerEntity = entityManager.HasComponent<WorkerEntityTag>(entity);
+            if (!hasSpatialEntityId && !isWorkerEntity)
+            {
+                worker.LogDispatcher.HandleLog(LogType.Warning, new LogEvent(
+                        "Attempted to link GameObject to an Entity that is not a SpatialOS Entity or the worker entity")
+                    .WithField(LoggingUtils.LoggerName, nameof(EntityGameObjectLinker)));
+                return;
+            }
+
+            EntityId spatialEntityId;
+            if (hasSpatialEntityId)
+            {
+                spatialEntityId = entityManager.GetComponentData<SpatialEntityId>(entity).EntityId;
+            }
+            else // worker entity
+            {
+                spatialEntityId = new EntityId(0);
```

Comment:
This shouldn't be hardcoded here, refactor out into a constant somewhere

Reply:
I'll make it a const field of the system